staging/src/k8s.io/cloud-provider/controllers/node/node_controller_test.go

```diff
@@ -924,9 +925,6 @@ func Test_syncNode(t *testing.T) {
 						},
 					},
 				},
-				Spec: v1.NodeSpec{
```

Comment:
@andrewsykim will also need to update the unit test for this change
if node providerID is not set, we will get the providerID by calling InstancesMetadata(), the fake response will return the metadata with providerID from `node.spec.providerID` (which is empty in this case)

Reply:
I think this is a bug in the fake cloud provider, where the providerID is not properly returned when using InstanceV2. We should fix that in a follow-up PR and update these tests accordingly.

Reply:
sure!